internal/httputil/certpoolwatcher.go

```diff
@@ -44,8 +45,38 @@ func NewCertPoolWatcher(caDir string, log logr.Logger) (*CertPoolWatcher, error)
 	if err != nil {
 		return nil, err
 	}
-	if err = watcher.Add(caDir); err != nil {
-		return nil, err
+
+	if caDir != "" {
+		// only watch if we can Stat() it
+		if _, err := os.Stat(caDir); err == nil {
+			if err = watcher.Add(caDir); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// If the SSL_CERT_DIR or SSL_CERT_FILE environment variables are
+	// specified, this means that we have some control over the system root
+	// location, thus they may change, thus we should watch those locations.
+	if d := os.Getenv("SSL_CERT_DIR"); d != "" {
+		dirs := strings.Split(d, ":")
```

Comment:
```suggestion
		dirs := filepath.SplitList(d)
```

Reply:
The separater is explicitly `":"`, and this is exactly how the golang library crypto/x509 does it, so I'm a bit reluctant to change it.

Reply:
Ok, if `crypto/x509` does it with simple `strings.Split`, I'm good.